types/underscore/index.d.ts

```diff
@@ -758,19 +760,23 @@ declare module _ {
 
         /**
          * Flattens a nested `list` (the nesting can be to any depth). If you
-         * pass shallow, the `list` will only be flattened a single level.
+         * pass true or 1 as the depth, the `list` will only be flattened a
+         * single level. Passing a greater number will cause the flattening to
+         * descend deeper into the nesting hierarchy. Omitting the depth
+         * argument, or passing false or Infinity, flattens the `list` all the
+         * way to the deepest nesting level.
          * @param list The list to flatten.
-         * @param shallow True to only flatten one level, optional,
+         * @param depth True to only flatten one level, optional,
          * default = false.
          * @returns The flattened `list`.
          **/
         flatten<V extends List<any>>(
             list: V,
-            shallow?: false
+            depth?: false
         ): DeepestListItemOrSelf<TypeOfList<V>>[];
         flatten<V extends List<any>>(
             list: V,
-            shallow: true
+            depth: number | true
         ): ListItemOrSelf<TypeOfList<V>>[];
```

Comment:
I suspect it is possible to be a bit more accurate than this, but I don't recall the exact details. Would be great if @reubenrybnik could comment on this.

Reply:
Ah you mean because when passing `true` or `1` as depth, the array will only be flattened a single level?

Reply:
Well, that case you already covered. But if `depth` is `2`, the return type is `ListItemOrSelf<ListItemOrSelf<TypeOfList<V>>>[]`. I think. I'm a bit rusty on the details.